pytorch_lightning/trainer/connectors/accelerator_connector.py

```diff
@@ -73,959 +71,762 @@
+    def _mapping_deprecated_devices_specfic_info_to_accelerator_and_device_flag(
+        self, devices, deprecated_devices_specific_flag, num_processes, gpus, ipus, tpu_cores
+    ):
+        # set devices base on num_processes, gpus, ipus, tpu_cores
+        if devices:
+            rank_zero_warn(
+                f"The flag `devices={devices}` will be ignored, "
+                f"instand the device specific number {deprecated_devices_specific_flag} will be used"
+            )
 
-    def _map_devices_to_accelerator(self, accelerator: str) -> bool:
-        if self.devices is None:
-            return False
-        if accelerator == _AcceleratorType.TPU and _TPU_AVAILABLE:
-            if self.devices == "auto":
-                self.devices = TPUAccelerator.auto_device_count()
-            self.tpu_cores = device_parser.parse_tpu_cores(self.devices)
-            return True
-        if accelerator == _AcceleratorType.IPU and _IPU_AVAILABLE:
-            if self.devices == "auto":
-                self.devices = IPUAccelerator.auto_device_count()
-            self.ipus = self.devices
-            return True
-        if accelerator == _AcceleratorType.GPU and torch.cuda.is_available():
-            if self.devices == "auto":
-                self.devices = GPUAccelerator.auto_device_count()
-            self.gpus = self.devices
-            self.parallel_device_ids = device_parser.parse_gpu_ids(self.devices)
-            return True
-        if accelerator == _AcceleratorType.CPU:
-            if self.devices == "auto":
-                self.devices = CPUAccelerator.auto_device_count()
-            if not isinstance(self.devices, int):
-                raise MisconfigurationException(
+        if [(num_processes is not None), (gpus is not None), (ipus is not None), (tpu_cores is not None)].count(
+            True
+        ) > 1:
+            rank_zero_warn("more than one device specifc flag has been set")
+        self._device_flag = deprecated_devices_specific_flag
+
+        if not self._accelerator_flag:
+            # set accelerator type base on num_processes, gpus, ipus, tpu_cores
+            if ipus:
+                self._accelerator_flag = "ipu"
+            if tpu_cores:
+                self._accelerator_flag = "tpu"
+            if gpus:
+                self._accelerator_flag = "gpu"
+            if num_processes:
+                self._accelerator_flag = "cpu"
+
+    def _choose_accelerator(self):
+        if self._accelerator_flag == "auto":
+            if _TPU_AVAILABLE:
+                self._accelerator_flag = "tpu"
+            elif _IPU_AVAILABLE:
+                self._accelerator_flag = "ipu"
+            elif torch.cuda.is_available() and torch.cuda.device_count() > 0:
+                self._accelerator_flag = "gpu"
+            else:
+                self._accelerator_flag = "cpu"
+                if self._device_flag == "auto":
+                    self._device_flag = 1
```

Comment:
why is the device check included here?
shouldn't this call CPUAccelerator.auto_device_count() instead of hardcoding?

Reply:
fixed